app/components/markdown/index.js

```diff
@@ -5,16 +5,16 @@ import {connect} from 'react-redux';
 
 import {getAutolinkedUrlSchemes, getConfig} from '@mm-redux/selectors/entities/general';
 import {getTheme} from '@mm-redux/selectors/entities/preferences';
-import {getCurrentUserMentionKeys} from '@mm-redux/selectors/entities/users';
+import {getAllUserMentionKeys} from '@mm-redux/selectors/entities/search';
 
 import Markdown from './markdown';
 
-function mapStateToProps(state) {
+function mapStateToProps(state, ownProps) {
     const {MinimumHashtagLength} = getConfig(state);
 
     return {
         autolinkedUrlSchemes: getAutolinkedUrlSchemes(state),
-        mentionKeys: getCurrentUserMentionKeys(state),
+        mentionKeys: ownProps.mentionKeys || getAllUserMentionKeys(state),
```

Comment:
Right. It's required all the way down to `at_mention.js` so I just need to pass `ownProps.mentionKeys`.

Reply:
Note, I realised why I'm doing this. If I just do `ownProps.mentionKeys`, the app fails.
The reason is because `<Markdown/>` is being called from other places other than `PostBody` so in those cases when they aren't passing `mentionKeys` as props, I need to retrieve all the users mentionKeys.